frameworks/Rust/actix-4/src/main_mongodb.rs

```diff
@@ -0,0 +1,215 @@
+mod models;
+
+use models::{Fortune, Queries, Result, World, CONNECTION_POOL_SIZE};
+
+use actix_http::{
+    header::{HeaderValue, CONTENT_TYPE, SERVER},
+    KeepAlive, StatusCode,
+};
+use actix_web::{web, App, HttpResponse, HttpServer};
+use anyhow::bail;
+use futures::{stream::FuturesUnordered, TryStreamExt};
+use mongodb::bson::doc;
+use mongodb::{options::ClientOptions, Client};
+use rand::{prelude::SmallRng, Rng, SeedableRng};
+use tokio::runtime::Handle;
+use yarte::ywrite_html;
+
+struct Data {
+    client: Client,
+    tokio_runtime: tokio::runtime::Handle,
+}
+
+async fn find_random_world(data: web::Data<Data>) -> Result<World> {
+    let runtime = data.tokio_runtime.clone();
```

Comment:
doing all the db work on the multithreded tokio runtime led to pretty big speedups here.